src/components/Footer/Footer.tsx

```diff
@@ -0,0 +1,89 @@
+import classNames from 'classnames';
+import { Todo } from '../../types/Todo';
+import { FilterType } from '../../utils/filterTypes';
+
+type Props = {
+  todos: Todo[];
+  filterType: FilterType;
+  onFilterTypeChange: (newFilterMode: FilterType) => void;
+};
+
+export const Footer: React.FC<Props> = ({
+  todos,
+  filterType,
+  onFilterTypeChange,
+}) => {
+  const activeTodos = todos.filter(todo => !todo.completed).length;
+  const completedTodos = todos.filter(todo => todo.completed).length;
+
+  return (
+    <footer className="todoapp__footer">
+      <span className="todo-count">
+        {`${activeTodos} items left`}
+      </span>
+
+      <nav className="filter">
+        <a
+          href="#/"
+          className={classNames(
+            'filter__link',
+            {
+              selected: filterType === FilterType.All,
+            },
+          )}
+          onClick={() => {
+            if (filterType !== FilterType.All) {
+              onFilterTypeChange(FilterType.All);
+            }
+          }}
```

Comment:
It's better to move such handlers into separate functions. Please, consider doing it everywhere.